src/spec.ts

```diff
@@ -161,6 +168,42 @@ export function normalizeUnitSpec(spec: UnitSpec): Spec {
   return spec;
 }
 
+export function normalizeCompositeUnitSpec(spec: UnitSpec): Spec {
+  let layerSpec = extend(spec.name ? {name: spec.name} : {},
+    spec.description ? {description: spec.description} : {},
+    spec.data ? {data: spec.data} : {},
+    spec.transform ? {transform: spec.transform} : {},
+    spec.config ? {config: spec.config} : {}, {layers: []});
+  if (!spec.encoding) {
+    return normalize(layerSpec);
```

Comment:
I think if there is no `encoding`, it shouldn't even enter this method from the beginning.